smoketest/calendarsubscription_test.go

```diff
@@ -0,0 +1,72 @@
+package smoketest
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/target/goalert/smoketest/harness"
+)
+
+func TestICal(t *testing.T) {
+	t.Parallel()
+
+	const sql = `
+	insert into users (id, name, email)
+	values
+		({{uuid "user"}}, 'bob', 'joe');
+	insert into schedules (id, name, time_zone, description) 
+	values
+		({{uuid "schedId"}},'sched', 'America/Chicago', 'test description here');
+	`
+	h := harness.NewHarness(t, sql, "calendar-subscriptions")
+	defer h.Close()
+
+	doQL := func(query string, res interface{}) {
+		g := h.GraphQLQuery2(query)
+		for _, err := range g.Errors {
+			t.Error("GraphQL Error:", err.Message)
+		}
+		if len(g.Errors) > 0 {
+			t.Fatal("errors returned from GraphQL")
+		}
+		t.Log("Response:", string(g.Data))
+		if res == nil {
+			return
+		}
+		err := json.Unmarshal(g.Data, &res)
+		if err != nil {
+			t.Fatal("failed to parse response:", err)
+		}
+	}
+
+	var cs struct{ CreateUserCalendarSubscription struct{ URL string } }
+
+	doQL(fmt.Sprintf(`
+		mutation {
+			createUserCalendarSubscription (input: {
+				name: "%s",
+				reminderMinutes: [%d]
+				scheduleID: "%s",
+			}) {
+				url
+			}
+		}
+	`, "foobar", 5, h.UUID("schedId")), &cs)
+
+	// get token
+	s := strings.Split(cs.CreateUserCalendarSubscription.URL, "token=")
```

Comment:
Doing this could double-encode the token value, and will break if any other params are present as `url.Values` is re-encoded before sending and splitting does not decode.

We can simplify it a bit though:
- parse as URL, assert no err, assert the path is `/api/v2/calendar`
- make the GET request using the original URL string

We just want to know, whatever string we get, works, and said string points to the path we expect. We should do some assertion on the contents and content-type of the response as well (e.g. make sure we actually get iCal data)